src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -205,6 +212,74 @@ public List<User> getAllUsersByGoogleId(String googleId) {
         return usersDb.getAllUsersByGoogleId(googleId);
     }
 
+    /**
+     * Common requirement between reset user by googleId function.
+     * Refer to {@link #resetInstructorGoogleId(String, String, String)} or
+     * {@link #resetStudentGoogleId(String, String, String)}.
+     */
+    private void deleteAccountCascadeAll(String googleId) {
+        if (usersDb.getAllInstructorsByGoogleId(googleId).isEmpty()
+                && usersDb.getAllStudentsByGoogleId(googleId).isEmpty()) {
+            accountsLogic.deleteAccountCascade(googleId);
+        }
+    }
+
+    /**
+     * Resets the googleId associated with the instructor.
+     */
+    public void resetInstructorGoogleId(String email, String courseId, String googleId)
+            throws EntityDoesNotExistException {
+        assert email != null;
+        assert courseId != null;
+        assert googleId != null;
+
+        Instructor instructor = getInstructorForEmail(courseId, email);
+
+        if (instructor == null) {
+            // Is there a better message?
+            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT
+                    + "Instructor [courseId=" + courseId + ", email=" + email + "]");
+        }
+
+        instructor.setAccount(null);
+
+        List<Account> accounts = accountsLogic.getAccountsForEmail(email);
+
+        for (Account account : accounts) {
+            accountsLogic.deleteAccount(account.getGoogleId());
+        }
+
+        deleteAccountCascadeAll(googleId);
```

Comment:
This is not correct. Many accounts are being deleted here. We only want to delete one account if it has been orphaned. Please take the time to understand the original behaviour of the action before performing the migration.

Reply:
Got it!